### ISSUE TITLE
CI failure in `MPResterOldTest.test_pourbaix_mpr_pipeline()`

### ISSUE THREAD

**Issue by janosh** [materialsproject/pymatgen#3039]
@rkingsbury `test_pourbaix_mpr_pipeline` has [started failing recently](https://github.com/materialsproject/pymatgen/actions/runs/5161568900/jobs/9298662810#step:7:31). Based on the call to Rester and the larger than expected number of `stable_entries`, is this because of new data in MP? If so, could we refactor this test to be more robust to future data additions?

```py
def test_pourbaix_mpr_pipeline(self):
        data = self.rester.get_pourbaix_entries(["Zn"])
        pbx = PourbaixDiagram(data, filter_solids=True, conc_dict={"Zn": 1e-8})
        pbx.find_stable_entry(10, 0)
    
        data = self.rester.get_pourbaix_entries(["Ag", "Te"])
        pbx = PourbaixDiagram(data, filter_solids=True, conc_dict={"Ag": 1e-8, "Te": 1e-8})
        assert len(pbx.stable_entries) == [30]
E       AssertionError: assert [29] == [30]

        test_entry = pbx.find_stable_entry(8, 2)
>       assert sorted(test_entry.entry_id) == ["ion-10", "mp-499"]
E       AssertionError: assert ['ion-10', 'mp-996958'] == ['ion-10', 'mp-499']
```

**Comment by janosh** [comment]
The odd thing is, I just ran this test locally and it passes.

```sh
pytest pymatgen/ext/tests/test_matproj.py -k test_pourbaix_mpr_pipeline
```

**Comment by rkingsbury** [comment]
Hi @janosh , I'm not sure what's happening here, but in general yes, changes to the database or API pipeline could potentially trigger test failures here. In this specific case, it looks like one entry has been dropped (reducing the count of stable entries), and a different entry (`mp-996958`) is stabilized than was before.

If `mp-499` were somehow excluded from the returned data, that might explain both failures. But weirdly, `mp-499` [still appears on MP](https://materialsproject.org/materials/mp-499?material_ids=mp-499) whereas `mp-996958` does not (it's not shown in search results, and manually [entering the URL](https://materialsproject.org/materials/mp-996958) will time out)

The test also passes for me when run locally.

I agree it would be nice to make this more robust to database changes, but I've had difficulty thinking of a good way to do so without making the test less rigorous. We could "dumb it down" such that we just test 1) that all the returned objects are `PourbaixEntry` type, or that there are a certain number of ions, but the total number of stable domains and the stable phases are really what define a Pourbaix Diagram, so I think it's nice to have a specific tests that verifies those.
